x-pack/plugins/security_solution/public/common/lib/local_storage/index.tsx

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Storage, IStorage } from '../../../../../../../src/plugins/kibana_utils/public';
+import { SecuritySolutionStorage } from './types';
+
+export const LOCAL_STORAGE_TIMELINE_KEY = 'timelines';
+
+export const createSecuritySolutionStorage = (store: IStorage): SecuritySolutionStorage => {
+  const storage = new Storage(store);
+
+  const getAllTimelines: SecuritySolutionStorage['getAllTimelines'] = () => {
+    return storage.get(LOCAL_STORAGE_TIMELINE_KEY);
+  };
+
+  const addTimeline: SecuritySolutionStorage['addTimeline'] = (id, timeline) => {
+    const timelines = getAllTimelines() ?? {};
```

Comment:
can we store `{}` as a variable to make sure that we avoid unnecessary re-renders? Because in the current state it will always return new object

Reply:
I think this logic will be better placed in getAllTimelines